File: CarCareHub_/CarCareHub.Model/AutoservisIzvjestaj.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CarCareHub.Model
{
    public class AutoservisIzvjestaj
    {
        public int AutoservisId { get; set; }
        public string NazivAutoservisa { get; set; }
        public decimal UkupanIznos { get; set; }
        public int BrojNarudzbi { get; set; }
        public decimal ProsjecnaCijena { get; set; }
        public List<ProizvodStatistika> NajpopularnijiProizvodi { get; set; } = new();
        public DateTime PeriodOd { get; set; } = DateTime.Now.AddDays(-30);
        public DateTime PeriodDo { get; set; } = DateTime.Now;
    }
}

File: CarCareHub_/UI_/flutter_mobile/lib/models/autoservisIzvjestaj.dart
import 'package:flutter_mobile/models/proizvodiStatistika.dart';
import 'package:json_annotation/json_annotation.dart';

part 'autoservisIzvjestaj.g.dart';

@JsonSerializable()
class AutoservisIzvjestaj {
  final int autoservisId;
  final String nazivAutoservisa;
  final double ukupanIznos;
  final int brojNarudzbi;
  final double prosjecnaCijena;
  final List<ProizvodStatistika> najpopularnijiProizvodi;
  final DateTime periodOd;
  final DateTime periodDo;

  AutoservisIzvjestaj({
    required this.autoservisId,
    required this.nazivAutoservisa,
    required this.ukupanIznos,
    required this.brojNarudzbi,
    required this.prosjecnaCijena,
    required this.najpopularnijiProizvodi,
    required this.periodOd,
    required this.periodDo,
  });

  factory AutoservisIzvjestaj.fromJson(Map<String, dynamic> json) =>
      _$AutoservisIzvjestajFromJson(json);

  Map<String, dynamic> toJson() => _$AutoservisIzvjestajToJson(this);
}
